### PR TITLE
DoNotFeedVeteranPets

### PR DESCRIPTION
**Pull request by Shadallark**
Since Veteran Pets should not have a feeding progress bar they should
not have the option to be fed either.

Added the same check used to remove their progress bar to the code to determine if the Feed option shows up on the ActionSheet.

### CODE REVIEW

File: HabitRPG/HRPGPetViewController.m [hunk 1]
```diff
@@ -143,7 +143,7 @@ - (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPa
     } else if ([self.equippedPetName isEqualToString:pet.key]) {
         equipString = NSLocalizedString(@"Unequip", nil);
     }
-    if (pet.asMount) {
+    if ((pet.asMount) || ([pet.type isEqualToString:@" "])){
```
Comment (crookedneighbor) [inline HabitRPG/HRPGPetViewController.m]: If it's the same check, would it make sense to pull it out into a function?

Comment (crookedneighbor) [inline HabitRPG/HRPGPetViewController.m]: @vIiRuS If so, add a unit test for it :wink:

**Comment by vIiRuS** [comment]
Yes, it's the same check. We could make a function for that, but the check if the type is " " is the only thing done.

What would the best way to test this be?

**Comment by crookedneighbor** [comment]
@vIiRuS I feel like that's a bad pattern, and really we should change the api so that type is meaningful for rare pets instead of a blank string. That'll break this and the other change, of course.

**Comment by crookedneighbor** [comment]
As far as testing goes, I'm not familiar with Obj C testing frameworks.

**Comment by Shadallark** [comment]
@crookedneighbor when you say change the API I am assuming that you mean an API that is being used by the website and all the different versions... sounds daunting... also seems necessary so that the type is meaningful for rare pets.

**Comment by vIiRuS** [comment]
@Shadallark yes. I will create a PR that changes the type to "rare", or something similar. I will have to check how the deployed app will handle that, but it ahouldn't have big problems with it. When the API PR is ready i will update here so that the pR can be asjusted.

**Comment by vIiRuS** [comment]
This actually isn't that trivial to change. The API doesn't contain any pet objects, but instead the displayed names come from the eggs. I'll have to change some things about the mapping to figure out a better way to map this. Or make a bigger change to the contents API. I'll merge this in for now and create a separate issue to solve the problem that the rare pets are identified by an " " type.

**Pull request merged**